src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -491,6 +493,26 @@ public function validateSchema(\DOMDocument $dom)
     }
 
     /**
+     * Validates an alias.
+     *
+     * @param \DOMElement
+     */
+    private function validateAlias(\DOMElement $alias, $file)
+    {
+        foreach (array('class', 'shared', 'synthetic', 'lazy', 'abstract', 'parent', 'decorates', 'decoration-inner-name', 'decoration-priority', 'autowire') as $key) {
+            if ($alias->getAttribute($key)) {
```

Comment:
Any reason `getAttribute()` is used here instead of [`hasAttribute()`](http://php.net/manual/en/domelement.hasattribute.php)?